### DISCUSSION TITLE
Metrics Generator span metrics to Prometheus without Prom Push Gateway

### DISCUSSION THREAD

**Discussion by andreivaloshyn** [grafana/tempo#2621] [Q&A] [answered]
Is there a way or something in the backlog to be able to have span metrics generated by metrics generator in Prometheus(without push gateway), something like serviceMonitor.
As I may see  - at this moment we have only remote writing configuration to do so https://grafana.com/docs/tempo/latest/metrics-generator/#remote-writing-metrics.

**Comment by joe-elliott** [comment, marked as answer]
This is not currently on the backlog and likely will not be prioritized by the team. Given the volume of metrics generated by Tempo remote write is generally a better choice then scraping.

However, you don't need push gateway to write span metrics from Tempo to Prometheus. Prometheus itself can be remote written to:
https://prometheus.io/docs/prometheus/latest/feature_flags/#remote-write-receiver

**Reply by andreivaloshyn** [reply]
@joe-elliott Thank you, it's exactly what I'm looking for.
And also looks like I need to improve my "googling" skill :)

**Comment by Charlotte-br560** [comment]
Currently, there isn't a direct integration for span metrics generated by Metrics Generator into Prometheus without using Push Gateway. It seems the only option available is through remote writing configuration as mentioned in the documentation. However, I'd recommend keeping an eye on the backlog for any updates
